linkerd/app/outbound/src/http/logical.rs

```diff
@@ -172,15 +173,76 @@ impl<E> Outbound<E> {
                         )
                         .into_inner(),
                 )
-                .push(profiles::http::NewServiceRouter::layer())
-                // Strips headers that may be set by this proxy and add an
-                // outbound canonical-dst-header. The response body is boxed
-                // unify the profile stack's response type with that of to
-                // endpoint stack.
-                .push(http::NewHeaderFromTarget::<CanonicalDstHeader, _>::layer())
-                .instrument(|l: &Logical| debug_span!("logical", dst = %l.logical_addr))
-                .push_on_service(svc::BoxService::layer())
-                .push(svc::ArcNewService::layer())
+                .push(profiles::http::NewServiceRouter::layer());
+
+            // for now, the client-policy route stack is just a fixed traffic split
+            let policy = concrete
+                .check_new_service::<(ConcreteAddr, Logical), _>()
+                .push_map_target(|(concrete, PolicyRoute { route: _, logical }): (ConcreteAddr, PolicyRoute)| {
+                    (concrete, logical)
+                })
+                .push(policy::split::layer())
```

Comment:
I may be wrong, but my reading of this means that the policy split creates an inner stack for every backend of every route. Does this mean that multiple routes to a single backend service do not share load balancers/connections? Or where are the load balancers cached if not here?

Reply:
Yes, that's correct; we should probably add a caching layer here so that each concrete address's client stack is shared across multiple routes that share that backend.

When we add more forms of per-route policy, though (e.g. different retry policies, HTTPRoute filters, and per route metrics), we'll want to ensure that the portions of that stack that may differ based on which route is used are before the client stack cache, though.